core/src/main/scala/org/apache/spark/ui/jobs/JobProgressListener.scala

```diff
@@ -93,6 +93,8 @@ class JobProgressListener(conf: SparkConf) extends SparkListener with Logging {
 
   val retainedStages = conf.getInt("spark.ui.retainedStages", SparkUI.DEFAULT_RETAINED_STAGES)
   val retainedJobs = conf.getInt("spark.ui.retainedJobs", SparkUI.DEFAULT_RETAINED_JOBS)
+  val retainedTasks = conf.getInt("spark.ui.retainedTasks", SparkUI.DEFAULT_RETAINED_TASKS)
```

Comment:
a long would just allow it to be set to a higher value.  2^64 vs 2^32.  Since the TaskId is a long already (see TaskInfo.scala) it makes sense to make the config a long. It really only matters if someone wants to set the limit over 2 billionish.

We have been trying to make any new configs the ConfigEntry type. so you don't have to do them all just the new one.  Its pretty straight forward, take a look at https://github.com/apache/spark/blob/master/core/src/main/scala/org/apache/spark/internal/config/package.scala